From cf618d03f707fc16178c303e64550604f787b270 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Fri, 8 Apr 2016 11:02:26 +0200
Subject: [PATCH] Ensure aodh and horizon templates do not overwrite ports.conf

If aodh is setup but nagios is not, the aodh.pp template could
overwrite the ports definition in Apache for Ceilometer and
Gnocchi.

Horizon could create a line in ports.conf for Ceilometer even if it
wasn't configured to run under httpd.

Change-Id: If9790f7a79f52bbad44a79a86b197df58d5da277
---
 packstack/puppet/templates/aodh.pp    | 12 ++++++++++++
 packstack/puppet/templates/horizon.pp |  4 +++-
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/packstack/puppet/templates/aodh.pp b/packstack/puppet/templates/aodh.pp
index d44afe2..90092c6 100644
--- a/packstack/puppet/templates/aodh.pp
+++ b/packstack/puppet/templates/aodh.pp
@@ -38,10 +38,22 @@ class { '::aodh::wsgi::apache':
   workers => $service_workers,
   ssl     => false
 }
+
 if hiera('CONFIG_KEYSTONE_SERVICE_NAME') == 'httpd' {
   apache::listen { '5000': }
   apache::listen { '35357': }
 }
+
+if hiera('CONFIG_GNOCCHI_INSTALL') == 'y' {
+  apache::listen { '8041': }
+}
+
+if hiera('CONFIG_CEILOMETER_INSTALL') == 'y' {
+  if hiera('CONFIG_CEILOMETER_SERVICE_NAME') == 'httpd' {
+    apache::listen { '8777': }
+  }
+}
+
 class { '::aodh::auth':
   auth_password => hiera('CONFIG_AODH_KS_PW'),
 }
diff --git a/packstack/puppet/templates/horizon.pp b/packstack/puppet/templates/horizon.pp
index 496fc09..ff56cb3 100644
--- a/packstack/puppet/templates/horizon.pp
+++ b/packstack/puppet/templates/horizon.pp
@@ -62,7 +62,9 @@ if hiera('CONFIG_KEYSTONE_SERVICE_NAME') == 'httpd' {
 }
 
 if hiera('CONFIG_CEILOMETER_INSTALL') == 'y' {
-  apache::listen { '8777': }
+  if hiera('CONFIG_CEILOMETER_SERVICE_NAME') == 'httpd' {
+    apache::listen { '8777': }
+  }
 }
 
 if hiera('CONFIG_AODH_INSTALL') == 'y' {
